DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,65 @@
+# Developer Guide
+
+So you want to contribute code to the OpenSearch Go Client?  Excellent!  We're glad you're here.  Here's what you need to do:
+
+## Getting Started
+
+### Git Clone OpenSearch Go Client Repository
+
+Fork [opensearch-project/opensearch-go](https://github.com/opensearch-project/opensearch-go) and clone locally,
+e.g. `git clone https://github.com/[your username]/opensearch-go.git`.
```

Comment:
@rushiagr link checker will try to look for https://github.com/[your username]/opensearch-go.git, hence, i have to exclude it.